src/integrationTest/java/uk/gov/hmcts/darts/event/controller/EventControllerDeleteEventMappingTest.java

```diff
@@ -0,0 +1,140 @@
+              "status": 403
+            }
+            """;
+
+        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+    @Test
+    void mappingInactive() throws Exception {
+        given.anAuthenticatedUserWithGlobalAccessAndRole(SUPER_ADMIN);
+        var entity = dartsDatabase.createEventHandlerData();
+        entity.setActive(false);
+        dartsDatabase.save(entity);
+        MockHttpServletRequestBuilder requestBuilder = delete(EVENT_MAPPINGS_ENDPOINT, entity.getId());
+        MvcResult response = mockMvc.perform(requestBuilder)
+            .andExpect(status().isConflict())
+            .andReturn();
+
+        String actualResponse = response.getResponse().getContentAsString();
+
+        String expectedResponse = """
+            {
+               "type": "EVENT_105",
+               "title": "The mapping is inactive, so cannot be deleted",
+               "status": 409,
+               "detail": "Event handler 379 cannot be deleted because it is inactive."
+            }
+            """;
+
+        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+
+    @Test
+    void mappingHasEvents() throws Exception {
+        given.anAuthenticatedUserWithGlobalAccessAndRole(SUPER_ADMIN);
+        LocalDateTime hearingDate = LocalDateTime.of(2020, 6, 6, 20, 0, 0);
+        HearingEntity hearing = dartsDatabase.createHearing("Courthouse", "1", "12345", hearingDate);
+        EventEntity event = dartsDatabase.createEvent(hearing);
+
+        MockHttpServletRequestBuilder requestBuilder = delete(EVENT_MAPPINGS_ENDPOINT, event.getEventType().getId());
+        MvcResult response = mockMvc.perform(requestBuilder)
+            .andExpect(status().isConflict())
+            .andReturn();
+
+        String actualResponse = response.getResponse().getContentAsString();
+
+        String expectedResponse = """
+            {
+              "type": "EVENT_106",
+              "title": "The mapping has already processed events, so cannot be deleted",
+              "status": 409,
+              "detail": "Event handler 10 already has processed events, so cannot be deleted."
+            }
+            """;
+
+        JSONAssert.assertEquals(expectedResponse, actualResponse, JSONCompareMode.NON_EXTENSIBLE);
+    }
+
+
+    @Test
+    void mappingDoesNotExist() throws Exception {
+        given.anAuthenticatedUserWithGlobalAccessAndRole(SUPER_ADMIN);
+
+        MockHttpServletRequestBuilder requestBuilder = delete(EVENT_MAPPINGS_ENDPOINT, -1);
+        MvcResult response = mockMvc.perform(requestBuilder)
+            .andExpect(status().isNotFound())
+            .andReturn();
+
+        String actualResponse = response.getResponse().getContentAsString();
+
+        String expectedResponse = """
+            {
+              "type":"EVENT_101",
+              "title": "No event handler found in database",
```

Comment:
Could we use the text 'No event handler _mapping_...' here, as per the pattern for the other endpoints? Event handler and event handler mapping are 2 different things.

Reply:
if this is the case, surely the event_handler table should be renamed?

Reply:
done